### PR TITLE
fix typos in topic overload policy

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/topic/TopicOverloadPolicy.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
  * This {@link TopicOverloadPolicy} controls how the publisher is going to deal with the situation that a ringbuffer is full
  *  and the oldest item in the ringbuffer is not old enough to get overwritten.
  *
- * So keep in mind that a this retention period keep the messages in memory, even though it might by that all readers already
+ * So keep in mind that this retention period keeps the messages in memory, even though it might be that all readers already
```
Comment (Serdaro) [inline hazelcast/src/main/java/com/hazelcast/topic/TopicOverloadPolicy.java]: So keep in mind that this retention period keeps the messages in memory, even though all readers may already

File: hazelcast/src/main/java/com/hazelcast/topic/TopicOverloadPolicy.java [hunk 2]
```diff
@@ -46,7 +46,7 @@
     DISCARD_OLDEST,
 
     /**
-     * The message that was to be published, is discarded.
+     * The message that was to be published, is discarded when there is no space in the ringbuffer.
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/topic/TopicOverloadPolicy.java]: All of the policies deal with the situation when there is no space in the ringbuffer.

**Comment by Serdaro** [comment]
@SaitTalhaNisanci , does this need further improvements?

**Comment by SaitTalhaNisanci** [comment]
@Serdaro  I think it doesnt need further improvements.

**Comment by mmedenjak** [comment]
@SaitTalhaNisanci does this PR need to go into 3.12? Can you rebase?

**Comment by mmedenjak** [comment]
Closing as this javadoc has already been rewritten in master.

**Pull request closed without merging**